### ISSUE TITLE
Add OTel Collector Security Best Practices to OTel Docs Site

### ISSUE THREAD

**Issue by mjingle** [open-telemetry/opentelemetry.io#3479]
**Desired feature or idea**: 

The OTel Collector repo contains a [Security Best Practices](https://github.com/open-telemetry/opentelemetry-collector/blob/main/docs/security-best-practices.md) document. This information is very useful and should be included in the OTel Docs site rather than hidden in a repo file.

Including this information is helpful in that it sets expectations, gives an opinion, and provides important security information to OTel Collector users.

**Additional context**:

I know the file itself is written with a repository lens, for example with mentions of specific issues, so depending on preference, some editorial work may be required before publishing.

Suggestion: to reduce maintenance, the repository file could be linked into the OTel docs site, so whenever the repository file is updated, the website version is also updated.

**Comment by cartermp** [comment]
Great idea - we should definitely have this documented on the site.

**Comment by chalin** [comment]
I'd rather not bring in another submodule if we can, especially for a single page. Would it be conceivable to have the page reside in this repo?

**Comment by cartermp** [comment]
I'd much prefer that, yes. @open-telemetry/collector-approvers any thoughts?

**Comment by mx-psi** [comment]
We discussed this previously, I think my comment on https://github.com/open-telemetry/opentelemetry.io/issues/3227#issuecomment-1711524397 still stands, IMO we should split the doc in two and have the user-focused part on the opentelemetry.io page

**Comment by mjingle** [comment]
To be clear, my suggestion to wire repos together is very much optional and intended to address potential concerns with maintainers keeping documentation up to date. I've encountered different preferences and solutions, but I totally understand the preference for less submodules.

I'm excited to see positive reception for this information to be in the public docs! Please let me know if there's anything I can do to keep moving this forward.

**Comment by svrnm** [comment]
@mjingle following @mx-psi's comment I think a good starting point would be taking a look at the current security documentation and extract what is user-facing and bringing it into a PR for the website.

**Comment by jpkrohling** [comment]
I agree that the _end user_ part of the linked documentation should be placed exclusively under opentelemetry.io. The current file in the repository could link to the website while keeping the advice targeted to component developers.

**Comment by theletterf** [comment]
@mjingle Could you advance with the draft?

**Comment by mx-psi** [comment]
It would be nice to have this completed for Collector v1 distro.

**Comment by svrnm** [comment]
@mjingle do you have time/bandwidth to work on this?

**Comment by tiffany76** [comment]
I've messaged @mjingle about this issue and to let her know I'm going to take over so we can get security documentation in place for v1.0. Hope to put up a fresh PR this week. Thanks!

**Comment by tiffany76** [comment]
@mx-psi, yes, that's exactly it. All that remains is to update the core repo README. I started work on that but got pulled into other things. I'll try to finish it up this week or next.
